### PR TITLE
2 step registration process for relayer

### PR DESCRIPTION
**Pull request by emailtovamos**

### CODE REVIEW

File: contracts/RelayerHub.sol [hunk 1]
```diff
@@ -129,28 +140,58 @@ contract RelayerHub is IRelayerHub, System, IParamSubscriber {
...
 
     function isRelayer(address relayerAddress) external override view returns (bool){
         return currentRelayers[relayerAddress];
     }
 
+    function isProvisionalRelayer(address relayerAddress) external view returns (bool){
+        return provisionalRelayers[relayerAddress];
+    }
+
     function isManager(address relayerAddress) external view returns (bool){
```
Comment (Mister-EA) [inline contracts/RelayerHub.sol:195]: Shouldn't the parameter name here be `address managerAddress`?

File: contracts/RelayerHub.sol [hunk 2]
```diff
@@ -129,28 +140,58 @@ contract RelayerHub is IRelayerHub, System, IParamSubscriber {
     }
 
     // updateRelayer() can be used to add relayer for the first time, update it in future and remove it
-    // in case of removal we can simply update it to a non-existing account
+    // in case of removal, we set relayerToBeAdded to be address(0)
     function updateRelayer(address relayerToBeAdded) public onlyManager {
         require(!isContract(relayerToBeAdded), "contract is not allowed to be a relayer");
 
         address oldRelayer = managerToRelayer[msg.sender];
```
Comment (guagualvcha) [inline contracts/RelayerHub.sol:147]: why not move `oldRelayer`, `oldProvisionalRelayer` inside the following `else` logic, to save gas.

File: contracts/RelayerHub.sol [hunk 3]
```diff
@@ -129,28 +140,58 @@ contract RelayerHub is IRelayerHub, System, IParamSubscriber {
     }
 
     // updateRelayer() can be used to add relayer for the first time, update it in future and remove it
-    // in case of removal we can simply update it to a non-existing account
+    // in case of removal, we set relayerToBeAdded to be address(0)
     function updateRelayer(address relayerToBeAdded) public onlyManager {
```
Comment (Mister-EA) [inline contracts/RelayerHub.sol:144]: Why not `external` ?

File: contracts/RelayerHub.sol [hunk 4]
```diff
@@ -129,28 +140,58 @@ contract RelayerHub is IRelayerHub, System, IParamSubscriber {
...
+        require(!isContract(msg.sender), "provisional relayer is a contract");
+        require(tx.origin == msg.sender, "provisional relayer is a proxy");
+        require(managerToProvisionalRelayer[manager] == msg.sender, "provisional is not set for this manager");
+
+        address oldRelayer = managerToRelayer[manager];
+
+        currentRelayers[msg.sender] = true;
+        managerToRelayer[manager] = msg.sender;
+
+        delete (provisionalRelayers[msg.sender]);
```
Comment (guagualvcha) [inline contracts/RelayerHub.sol:180]: Please delete manager in managerToProvisionalRelayer as well

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**